File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    # 가로 : n
    # 세로 : m
    # brown : 2 * (n + m - 2)
    # yellow : (n - 2) * (m - 2)
    
    # n = 4
    # m = 3
    # b = 2 * (n + m - 2)
    # y = (n - 2) * (m - 2)
    # return (b == brown and y == yellow)

    for n in range(brown):
        for m in range(brown):
            b = 2 * (n + m - 2)
            y = (n - 2) * (m - 2)
            if b == brown and y == yellow:
                return [max(n, m), min(n, m)]
    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer = {-1, -1};
    // yellow = n * m
    // all = (n+2) * (m+2)
    // brown = all - yellow
    // ans => [n+2, m+2]
    
    for(int n = 1; n <= brown; n++)
    {
        for(int m = 1; m <= brown; m++)
        {
            if (yellow != m * n) continue;            
            if (brown != ((n+2) * (m+2) - (n * m))) continue;
                        
            int max = n < m ? m : n;
            int mim = n > m ? m : n;
            answer = {max + 2, mim + 2};
            return answer;
        }
    }
            
    return answer;
}